go/vt/vttablet/tabletserver/throttle/throttler_test.go

```diff
@@ -75,3 +116,93 @@ func TestIsAppExempted(t *testing.T) {
 	throttler.UnthrottleApp("schema-tracker") // meaningless. App is statically exempted
 	assert.True(t, throttler.IsAppExempted("schema-tracker"))
 }
+
+func TestRefreshMySQLInventory(t *testing.T) {
+	metricsQuery := "select 1"
+	config.Settings().Stores.MySQL.Clusters = map[string]*config.MySQLClusterConfigurationSettings{
+		selfStoreName: {},
+		"ks1":         {},
+		"ks2":         {},
+	}
+	clusters := config.Settings().Stores.MySQL.Clusters
+	for _, s := range clusters {
+		s.MetricQuery = metricsQuery
+		s.ThrottleThreshold = &atomic.Uint64{}
+		s.ThrottleThreshold.Store(1)
+	}
+
+	throttler := &Throttler{
+		mysqlClusterProbesChan: make(chan *mysql.ClusterProbes),
+		mysqlClusterThresholds: cache.New(cache.NoExpiration, 0),
+		ts:                     &FakeTopoServer{},
+		mysqlInventory:         mysql.NewInventory(),
+	}
+	throttler.metricsQuery.Store(metricsQuery)
+	throttler.initThrottleTabletTypes()
+
+	validateClusterProbes := func(t *testing.T, ctx context.Context) {
+		testName := fmt.Sprintf("leader=%t", throttler.isLeader.Load())
+		t.Run(testName, func(t *testing.T) {
+			t.Run("waiting for probes", func(t *testing.T) {
+				ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+				defer cancel()
+				numClusterProbesResults := 0
+				for {
+					select {
+					case probes := <-throttler.mysqlClusterProbesChan:
+						assert.NotNil(t, probes)
+
+						throttler.updateMySQLClusterProbes(ctx, probes)
+
+						numClusterProbesResults++
+						if probes.ClusterName == selfStoreName {
+							assert.Equal(t, 1, len(*probes.InstanceProbes))
+						} else if throttler.isLeader.Load() {
+							assert.NotZero(t, len(*probes.InstanceProbes))
+						} else {
+							assert.Empty(t, *probes.InstanceProbes)
+						}
+						if numClusterProbesResults == len(clusters) {
+							return
+						}
+					case <-ctx.Done():
+						assert.FailNowf(t, ctx.Err().Error(), "waiting for %d cluster probes", len(clusters))
+					}
+				}
+			})
+			t.Run("validating probes", func(t *testing.T) {
+				for clusterName := range clusters {
+					probes, ok := throttler.mysqlInventory.ClustersProbes[clusterName]
+					require.True(t, ok)
+					if clusterName == selfStoreName {
+						assert.Equal(t, 1, len(*probes))
+					} else if throttler.isLeader.Load() {
+						assert.NotZero(t, len(*probes))
+					} else {
+						assert.Empty(t, *probes)
+					}
```

Comment:
Isn't this repeating the same logic from the wait? Fine to keep it in, but I'm not sure it's needed.